### PR TITLE
plot histogram with class bins

### PR DESCRIPTION
**Pull request by knaaptime**
wanted to gauge interest in adding a method to Classifier that plots the [histogram of `y` with class bins](https://knaaptime.com/urban_analysis/01_geoprocessing/01_geovisualization.html#fig-dist-cuts) a la desktop GIS

Screenshot 2024-06-18 at 9 23 12 PM

pandas and seaborn are optional dependencies but usually available in a pysal environment. If we take the coloring logic from #211 I think the [legendgram](https://github.com/pysal/legendgram/tree/master) could also be a method on Classifier (without depending on palettable), which would be pretty handy. Will add a test if folks are interested

### CODE REVIEW

**Comment by martinfleis** [comment]
Nice, I like this and the idea of moving legendgram as well.

I would probably just try to add only maptlotlib as optional dependency. You can easily use `plt.hist` instead of using pandas and if seaborn is needed only for a minor cosmetic change, you can easily call `sns.despine()` yourself.

**Comment by martinfleis** [comment]
I don't think we need `seaborn` to remove spines :).

```py
ax.spines.right.set_visible(False)
ax.spines.top.set_visible(False)
```

from https://matplotlib.org/3.4.3/gallery/ticks_and_spines/spines_dropped.html

**Comment by knaaptime** [comment]
oh right :P thanks

**Comment by knaaptime** [comment]
plt.hist doesnt take or return an axes though (so the ax.vline doesn't work and isn't as composable otherwise). I think I'll keep it as pandas unless you have a different implementation?

**Comment by martinfleis** [comment]
Like this

```py
rng = np.random.default_rng(seed=0)
normal = rng.normal(size=1000)

fig, ax = plt.subplots()

ax.hist(normal)

lim = ax.get_ylim()[1]
ax.vlines(-1, 0, lim, color="k")
ax.spines.right.set_visible(False)
ax.spines.top.set_visible(False)
```

File: mapclassify/classifiers.py [hunk 1]
```diff
@@ -1123,6 +1123,68 @@ def plot(
             plt.savefig(file_name, dpi=dpi)
         return f, ax
 
+    def plot_histogram(
+        self,
+        hist_color="dodgerblue",
```
Comment (martinfleis) [inline mapclassify/classifiers.py:1128]: Shall we just simply call this `color`?

File: mapclassify/classifiers.py [hunk 2]
```diff
@@ -1123,6 +1123,68 @@ def plot(
...
+        linecolor="black",
+        ax=None,
+        despine=True,
+        hist_kwargs=None,
```
Comment (martinfleis) [inline mapclassify/classifiers.py:1132]: Could this just be `**kwargs` so you don't have to define a dict when passing additional values?

File: mapclassify/classifiers.py (hunk 1 above)
Comment (knaaptime) [reply mapclassify/classifiers.py:1128]: makes sense to me. Wasnt sure if we should differentiate between the bar color and the line color. Seems reasonable that 'color' defaults to the bar color

File: mapclassify/classifiers.py (hunk 2 above)
Comment (knaaptime) [reply mapclassify/classifiers.py:1132]: yup

**Review by jGaboardi — APPROVED** [review]
fancy!

**Pull request merged**